bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/io/ChannelPool.java

```diff
@@ -57,26 +58,55 @@
    */
   protected static Counter ACTIVE_CHANNEL_COUNTER;
 
+  /**
+   * A factory for creating ManagedChannels to be used in a {@link ChannelPool}.
+   *
+   * @author sduskis
+   *
+   */
+  public interface ChannelFactory {
+    ManagedChannel create() throws IOException;
+  }
+
+
+  /**
+   * There's a classloading vs. user configuration timing issue. A user should have the opportunity
+   * to configure metrics gathering even if the class loader loads the ChannelPool. If a user turns
+   * on metrics after this class is loaded, these metrics should not have the NULL
+   * implementation.
+   * @return
+   */
   protected synchronized static Counter getActiveChannelCounter() {
     if (ACTIVE_CHANNEL_COUNTER == null) {
       ACTIVE_CHANNEL_COUNTER = BigtableClientMetrics.createCounter("ChannelPool.active.count");
     }
     return ACTIVE_CHANNEL_COUNTER;
   }
 
-  public interface ChannelFactory {
-    ManagedChannel create() throws IOException;
-  }
-
+  /**
+   * Contains a {@link ManagedChannel} and metrics for the channel
```

Comment:
Sorry for not asking this earlier, but did you consider subclassing ManagedChannel so that this is a real wrapper?

If you did that you could roll in the channelId concept, and fold in the markInactive semantics to the shutdown/shutdownNow overrides.